In [1]:
from helpers.environment import Environment, EnvironmentType
from helpers.device import Device, Motor, AirConditioned, Light, LightType
from helpers.installation import Installation, InstallationType


environments = [
	Environment(
		"Escritório 1",
		2.30,
		3.9,
		EnvironmentType.OFFICES_AND_CLASSROOMS,
		Light(15, LightType.LED),
		[AirConditioned("Ar condicionado escritório 1", 2000)]
	),

	Environment(
		"Escritório 2",
		2.70,
		5.10,
		EnvironmentType.OFFICES_AND_CLASSROOMS,
		Light(15, LightType.LED),
		[AirConditioned("Ar condicionado escritório 2", 2000)]
	),

	Environment(
		"Recepção",
		2.80,
		5.10,
		EnvironmentType.WAITING_ROOMS_AND_COMMUNAL_AREAS,
		Light(15, LightType.LED),
		[AirConditioned("Ar condicionado recepção", 2000)]
	),

	Environment(
		"Vestiário",
		2.25,
		2.70,
		EnvironmentType.BATHROOMS_AND_OTHER_FACILITIES,
		Light(15, LightType.LED),
		[]
	),

	Environment(
		"Cozinha",
		2.70,
		1.90,
		EnvironmentType.KITCHENS_AND_DINING_ROOMS,
		Light(15, LightType.LED),
		[Device("Microondas", 2000, 0.92), Device("Chaleira", 1200, 1.0)]
	),

	Environment(
		"Refeitório",
		2.70,
		2.90,
		EnvironmentType.RESTAURANTS_CAFETERIAS_BARS_AND_HOTELS,
		Light(15, LightType.LED),
		[]
	),

	Environment(
		"Depósito",
		2.70,
		1.30,
		EnvironmentType.WAREHOUSES_DEPOSITS_AND_STORAGE_AREAS,
		Light(15, LightType.LED),
		[]
	),

	Environment(
		"Banheiro",
		1.95,
		1.2,
		EnvironmentType.BATHROOMS_AND_OTHER_FACILITIES,
		Light(15, LightType.LED),
		[Device("Chapinha / Secador", 1200, 1.0)]
	),

	Environment(
		"Banho",
		1.99562,
		1.99562,
		EnvironmentType.BATHROOMS_AND_OTHER_FACILITIES,
		Light(15, LightType.LED),
		[Device("Chuveiro", 7000, 1.0), Device("Chapinha / Secador", 1200, 1.0)]
	),

	Environment(
		"Supervisão de Produção",
		3.28633,
		3.28633,
		EnvironmentType.OFFICES_AND_CLASSROOMS,
		Light(15, LightType.LED),
		[]
	),

	Environment(
		"Produção 1",
		7.5,
		8,
		EnvironmentType.LAUNDRIES_AND_WORKSHOPS,
		Light(15, LightType.LED),
		[Motor(horse_power = 5.0, efficiency = 0.92, name = "Batedor de argila")] +
		6 * [Motor(name = "Molde e cortador", horse_power = 1.0, power_factor = 0.85, efficiency = 0.95)]
	),

	Environment(
		"Produção 2",
		7.5,
		8,
		EnvironmentType.LAUNDRIES_AND_WORKSHOPS,
		Light(15, LightType.LED),
		2 * [Motor(horse_power = 2.0, power_factor = 0.8, efficiency = 0.95)] +
		[Motor(horse_power = 2.0, power_factor = 0.83, efficiency = 0.93)] +
		[Device(name = "Forno", power = 6000, power_factor = 1.0)]
	),

	Environment(
		"Produção 3",
		4,
		10.4,
		EnvironmentType.LAUNDRIES_AND_WORKSHOPS,
		Light(15, LightType.LED),
		2 * [Motor(horse_power = 1.0, power_factor = 0.85, efficiency = 0.95)]
	),

	Environment(
		"Empacotamento e Carga e descarga",
		4,
		10.4,
		EnvironmentType.LAUNDRIES_AND_WORKSHOPS,
		Light(15, LightType.LED),
		2 * [Motor(horse_power = 1.0, power_factor = 0.75, efficiency = 0.90)]
	),

]

installation = Installation(environments, InstallationType.THREE_PHASE)

print(installation)


Installation:
	Light Power: 4172.88W
	Light Number: 28
	Tug Power: 18700W
	Tug Number: 69
	Tue Power: 40266.0W
	Tue Number: 23
	Total Demanded Power: 21193.23W
	Installation Type: InstallationType.THREE_PHASE
	Phases: 3
	Total Demanded Apparent Power: 22308.66VA
	Total Demanded Current: 32.11A
	Input Branch Current: 40.14A
Enviroment:
	Name: Escritório 1 - Area: 8.97m² - Perimeter: 12.4m
	Light:
		Light: Lampada - 15 W - PF: 0.65 - 23.08 VA
		Recommended Power: 107.64W
		Minimum Light Power: 100W
		Light Power: 107.64W
		Minimum Light Points: 4
	Tug:
		Minimum Tug Number: 4
		Minimum Tug Power: 800W
	Tue:
		Minimum Tue Number: 1
Enviroment:
	Name: Escritório 2 - Area: 13.77m² - Perimeter: 15.6m
	Light:
		Light: Lampada - 15 W - PF: 0.65 - 23.08 VA
		Recommended Power: 165.24W
		Minimum Light Power: 160W
		Light Power: 165.24W
		Minimum Light Points: 4
	Tug:
		Minimum Tug Number: 5
		Minimum Tug Power: 1000W
	Tue:
		Minimum Tue Number: 1
Enviroment:
	Name: Recepção - Area: 14.28m² - Per

# Separação dos Circuitos

---

## Quadro Geral



## Quadro terminal 1

- **Número de disjuntores**: 17 (13 + 4 Reservas)
- **Número de TUEs**: 7
- **Poência demandada TUG + LUZ**: 3080W
- **Potência demandada TUE**: 12616W
- **Potência demandada Total**: (3080 * 0.59) + (12616 * 0.6) = 9387W (considerando 9500W)

### Escritório 1

- **Número de disjuntores**: 3
- **Número de TUEs**: 1
- **Poência demandada TUG + LUZ**: 930W
- **Potência demandada TUE**: 2000W

- **Disjuntor 1 (Iluminação)**: 15A
  - Condutor: 1.5mm² (capacidade de 17.5A)
  - 4 pontos de iluminação
  - Potência requerida 100VA (considerando 200VA)
  - Potência ativa nominal: LED(FP: 0.65) 130W

- **Disjuntor 2 (Tomadas TUG)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 4 pontos de utilização
  - Potência requerida 800VA (considerando 1000VA)
  - Potência ativa nominal: TUG(FP: 0.8) 800W

- **Disjuntor 3 (Tomadas TUE)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 1 ponto de utilização (específico)
  - Potência requerida 2000VA (considerando 2500VA)
  - Potência ativa nominal: (FP/REND: 0.8) 2000W

### Escritório 2

- **Número de disjuntores**: 3
- **Número de TUEs**: 1
- **Poência demandada TUG + LUZ**: 1090W
- **Potência demandada TUE**: 2000W

- **Disjuntor 4 (Iluminação)**: 15A
  - Condutor: 1.5mm² (capacidade de 17.5A)
  - 4 pontos de iluminação
  - Potência requerida 100VA (considerando 200VA)
  - Potência ativa nominal: LED(FP: 0.65) 130W

- **Disjuntor 5 (Tomadas TUG)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 5 pontos de utilização
  - Potência requerida 1000VA (considerando 1200VA)
  - Potência ativa nominal: TUG(FP: 0.8) 960W

- **Disjuntor 6 (Tomadas TUE)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 1 ponto de utilização (específico)
  - Potência requerida 2000VA (considerando 2500VA)
  - Potência ativa nominal: (FP/REND: 0.8) 2000W

### Recepção + Depósito

- **Número de disjuntores**: 3
- **Número de TUEs**: 1
- **Poência demandada TUG + LUZ**: 1060W
- **Potência demandada TUE**: 2000W

- **Disjuntor 7 (Iluminação)**: 15A
  - Condutor: 1.5mm² (capacidade de 17.5A)
  - 2 pontos de iluminação na Recepção + 2 pontos no Depósito
  - Potência requerida 320VA (considerando 400VA)
  - Potência ativa nominal: LED(FP: 0.65) 260W

- **Disjuntor 8 (Tomadas TUG)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 5 pontos de utilização na Recepção + 2 pontos no Depósito
  - Potência requerida 700VA (considerando 1000VA)
  - Potência ativa nominal: TUG(FP: 0.8) 800W

- **Disjuntor 9 (Tomadas TUE - Recepção)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 1 ponto de utilização (específico)
  - Potência requerida 2000VA (considerando 2500VA)
  - Potência ativa nominal: (FP/REND: 0.8) 2000W

### Banho + Banheiro (TUE balanceamento de fase)

- **Número de disjuntores**: 2
- **Número de TUEs**: 2
- **Poência demandada TUG + LUZ**: 0W
- **Potência demandada TUE**: 3000W

- **Disjuntor 8 (Tomadas TUE Chapinha / Secador)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 1 ponto de utilização no Banheiro (específico)
  - Potência requerida 1200VA (considerando 1500VA)
  - Potência ativa nominal: (FP/REND: 1.0) 1500W

- **Disjuntor 9 (Tomadas TUE Chapinha / Secador)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 1 ponto de utilização no Banho (específico)
  - Potência requerida 1200VA (considerando 1500VA)
  - Potência ativa nominal: (FP/REND: 1.0) 1500W

### Cozinha (TUE balanceamento de fase)

- **Número de disjuntores**: 2
- **Número de TUEs**: 2
- **Poência demandada TUG + LUZ**: 0W
- **Potência demandada TUE**: 3616W

- **Disjuntor 4 (Tomadas TUE Cozinha Microondas)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 1 ponto de utilização na Cozinha (específico)
  - Potência requerida 2000VA (considerando 2300VA)
  - Potência ativa nominal: (FP/REND: 0.92) 2116W

- **Disjuntor 5 (Tomadas TUE Cozinha Chaleira)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 1 ponto de utilização no Refeitório (específico)
  - Potência requerida 1200VA (considerando 1500VA)
  - Potência ativa nominal: (FP/REND: 1.0) 1500W

---

## Quadro terminal 2

- **Número de disjuntores**: 11 (8 + 3 Reservas)
- **Número de TUEs**: 1
- **Poência demandada TUG + LUZ**: 6270W
- **Potência demandada TUE**: 13616W
- **Potência demandada Total**: (6270 * 0.4) + (7000 * 1.0) = 9509W (considerando 9600W)

### Cozinha + Refeitório

- **Número de disjuntores**: 3
- **Número de TUEs**: 0
- **Poência demandada TUG + LUZ**: 2200W
- **Potência demandada TUE**: 0W

- **Disjuntor 1 (Iluminação)**: 15A
  - Condutor: 1.5mm² (capacidade de 17.5A)
  - 2 pontos de iluminação na Cozinha + 4 pontos no Refeitório
  - Potência requerida 200VA (considerando 300VA)
  - Potência ativa nominal: LED(FP: 0.65) 200W

- **Disjuntor 2 (Tomadas TUG Cozinha)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 3 pontos de utilização na Cozinha
  - Potência requerida 1800VA (considerando 2000VA)
  - Potência ativa nominal: TUG(FP: 0.8) 1600W

- **Disjuntor 3 (Tomadas TUG Refeitório)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)****
  - 4 pontos de utilização no Refeitório
  - Potência requerida 400VA (considerando 500VA)
  - Potência ativa nominal: TUG(FP: 0.8) 400W

### Vestiário + Banheiro + Banho

- **Número de disjuntores**: 3
- **Número de TUEs**: 1
- **Poência demandada TUG + LUZ**: 3140W
- **Potência demandada TUE**: 7000W

- **Disjuntor 6 (Iluminação)**: 15A
  - Condutor: 1.5mm² (capacidade de 17.5A)
  - 2 pontos de iluminação no Vestiário + 2 pontos no Banheiro + 2 ponto no Banho
  - Potência requerida 300VA (considerando 400VA)
  - Potência ativa nominal: LED(FP: 0.65) 260W

- **Disjuntor 7 (Tomadas TUG)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 2 pontos de iluminação no Vestiário + 2 pontos no Banheiro + 2 ponto no Banho
  - Potência requerida 3600VA (considerando 3600VA)
  - Potência ativa nominal: TUG(FP: 0.8) 2880W

- **Disjuntor 10 (Tomadas TUE Chuveiro)**: 35A
  - Condutor: 6.0mm² (capacidade de 41A)
  - 1 ponto de utilização no Banho (específico)
  - Potência requerida 7000VA (considerando 7000VA)
  - Potência ativa nominal: (FP/REND: 1.0) 7000W

### Supervisão de produção

- **Número de disjuntores**: 2
- **Número de TUEs**: 0
- **Poência demandada TUG + LUZ**: 930W
- **Potência demandada TUE**: 0W

- **Disjuntor 11 (Iluminação)**: 15A
  - Condutor: 1.5mm² (capacidade de 17.5A)
  - 4 pontos de iluminação
  - Potência requerida 160VA (considerando 200VA)
  - Potência ativa nominal: LED(FP: 0.65) 130W

- **Disjuntor 12 (Tomadas TUG)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 4 pontos de utilização
  - Potência requerida 800VA (considerando 1000VA)
  - Potência ativa nominal: TUG(FP: 0.8) 800W

---

## Quadro terminal 3

- **Número de disjuntores**: 17 (13 + 4 Reservas)
- **Número de TUEs**: 1
- **Poência demandada TUG + LUZ**: 12144W
- **Potência demandada TUE**: 6500W
- **Potência demandada Total**: (12144 * 0.24) + (6500 * 1.0) = 9415W (considerando 9600W)

### Produção 1

- **Número de disjuntores**: 3
- **Número de TUEs**: 0
- **Poência demandada TUG + LUZ**: 3224W
- **Potência demandada TUE**: 0W

- **Disjuntor 1 (Iluminação)**: 15A
  - Condutor: 1.5mm² (capacidade de 17.5A)
  - 4 pontos de iluminação na Produção 1 (Cada ponto é uma luminária dupla)
  - Potência requerida 880VA (considerando 900VA)
  - Potência ativa nominal: FLUOR(fp: 0.5 / Preact: 4.8) 1064W

- **Disjuntor 2 (Tomadas TUG 1)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 3 pontos de utilização
  - Potência requerida 1800VA (considerando 2000VA)
  - Potência ativa nominal: TUG(FP: 0.8) 1600W

- **Disjuntor 3 (Tomadas TUG 2)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 6 pontos de utilização
  - Potência requerida 600VA (considerando 700VA)
  - Potência ativa nominal: TUG(FP: 0.8) 560W

### Produção 2

- **Número de disjuntores**: 4
- **Número de TUEs**: 1
- **Poência demandada TUG + LUZ**: 3224W
- **Potência demandada TUE**: 6500W

- **Disjuntor 1 (Iluminação)**: 15A
  - Condutor: 1.5mm² (capacidade de 17.5A)
  - 4 pontos de iluminação na Produção 2 (Cada ponto é uma luminária dupla)
  - Potência requerida 880VA (considerando 900VA)
  - Potência ativa nominal: FLUOR(fp: 0.5 / Preact: 4.8) 1064W

- **Disjuntor 2 (Tomadas TUG 1)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 3 pontos de utilização
  - Potência requerida 1800VA (considerando 2000VA)
  - Potência ativa nominal: TUG(FP: 0.8) 1600W

- **Disjuntor 3 (Tomadas TUG 2)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 6 pontos de utilização
  - Potência requerida 600VA (considerando 700VA)
  - Potência ativa nominal: TUG(FP: 0.8) 560W

- **Disjuntor 4 (Tomadas TUE Forno)**: 32A
  - Condutor: 6.0mm² (capacidade de 41A)
  - 1 ponto de utilização na Produção 2 (específico)
  - Potência requerida 6000VA (considerando 6500VA)
  - Potência ativa nominal: (FP/REND: 1.0) 6500W

### Produção 3

- **Número de disjuntores**: 3
- **Número de TUEs**: 0
- **Poência demandada TUG + LUZ**: 2848W
- **Potência demandada TUE**: 0W

- **Disjuntor 1 (Iluminação)**: 15A
  - Condutor: 1.5mm² (capacidade de 17.5A)
  - 4 pontos de iluminação na Produção 3 (Cada ponto é uma luminária dupla)
  - Potência requerida 580VA (considerando 600VA)
  - Potência ativa nominal: FLUOR(fp: 0.5 / Preact: 4.8) 688W

- **Disjuntor 2 (Tomadas TUG 1)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 3 pontos de utilização
  - Potência requerida 1800VA (considerando 2000VA)
  - Potência ativa nominal: TUG(FP: 0.8) 1600W

- **Disjuntor 3 (Tomadas TUG 2)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 6 pontos de utilização
  - Potência requerida 600VA (considerando 700VA)
  - Potência ativa nominal: TUG(FP: 0.8) 560W

### Empacotamento e Carga e descarga

- **Número de disjuntores**: 3
- **Número de TUEs**: 0
- **Poência demandada TUG + LUZ**: 2848W
- **Potência demandada TUE**: 0W

- **Disjuntor 1 (Iluminação)**: 15A
  - Condutor: 1.5mm² (capacidade de 17.5A)
  - 6 pontos de iluminação (Cada ponto é uma luminária dupla)
  - Potência requerida 580VA (considerando 600VA)
  - Potência ativa nominal: FLUOR(fp: 0.5 / Preact: 4.8) 688W

- **Disjuntor 2 (Tomadas TUG 1)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 3 pontos de utilização
  - Potência requerida 1800VA (considerando 2000VA)
  - Potência ativa nominal: TUG(FP: 0.8) 1600W

- **Disjuntor 3 (Tomadas TUG 2)**: 20A
  - Condutor: 2.5mm² (capacidade de 24A)
  - 6 pontos de utilização
  - Potência requerida 600VA (considerando 700VA)
  - Potência ativa nominal: TUG(FP: 0.8) 560W

---
